starknet_py/net/client_models.py

```diff
@@ -191,28 +193,53 @@ def __post_init__(self):
 
 
 @dataclass
-class InvokeTransaction(DeprecatedTransaction):
```

Comment:
Just like in the schemas, let's keep `DeprecatedTransaction`, `TransactionV3` and have only version specifc classes, such as `InvokeTransactionV0`, `InvokeTransactionV1` etc. We'd like to avoid having classes with generic names like `InvokeTransaction` or `DeclareTransaction`. These names imply they are default classes for given transactions, which isn't true. We may consider introducing later something like `InvokeTransactionCommonFields`, `DeclareTransactionCommonFields` etc.